lib/ansible/modules/database/postgresql/postgresql_publication.py

```diff
@@ -0,0 +1,652 @@
+        """
+        if self.exists:
+            query_fragments = []
+            query_fragments.append("DROP PUBLICATION %s" % pg_quote_identifier(self.name, 'publication'))
+            if cascade:
+                query_fragments.append("CASCADE")
+
+            if check_mode:
+                self.executed_queries.append(' '.join(query_fragments))
+                return True
+            else:
+                return exec_sql(self, ' '.join(query_fragments), ddl=True)
+
+    def __get_general_pub_info(self):
+        """Get and return general publication information.
+
+        Returns:
+            Dict with publication information if successful, False otherwise.
+        """
+        # Check pg_publication.pubtruncate exists (supported from PostgreSQL 11):
+        pgtrunc_sup = exec_sql(self, ("SELECT 1 FROM information_schema.columns "
+                                      "WHERE table_name = 'pg_publication' "
+                                      "AND column_name = 'pubtruncate'"), add_to_executed=False)
+
+        if pgtrunc_sup:
+            query = ("SELECT r.rolname AS pubowner, p.puballtables, p.pubinsert, "
+                     "p.pubupdate , p.pubdelete, p.pubtruncate FROM pg_publication AS p "
+                     "JOIN pg_catalog.pg_roles AS r "
+                     "ON p.pubowner = r.oid "
+                     "WHERE p.pubname = '%s'" % self.name)
+        else:
+            query = ("SELECT r.rolname AS pubowner, p.puballtables, p.pubinsert, "
+                     "p.pubupdate , p.pubdelete FROM pg_publication AS p "
+                     "JOIN pg_catalog.pg_roles AS r "
+                     "ON p.pubowner = r.oid "
+                     "WHERE p.pubname = '%s'" % self.name)
+
+        result = exec_sql(self, query, add_to_executed=False)
+        if result:
+            return result[0]
+        else:
+            return False
+
+    def __get_tables_pub_info(self):
+        """Get and return tables that are published by the publication.
+
+        Returns:
+            List of dicts with published tables.
+        """
+        query = ("SELECT schemaname, tablename "
+                 "FROM pg_publication_tables WHERE pubname = '%s'" % self.name)
+        return exec_sql(self, query, add_to_executed=False)
+
+    def __pub_add_table(self, table, check_mode=False):
+        """Add a table to the publication.
+
+        Args:
+            table (str): Table name.
+
+        Kwargs:
+            check_mode (bool): If True, don't actually change anything,
+                just make SQL, add it to ``self.executed_queries`` and return True.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+        query = ("ALTER PUBLICATION %s ADD TABLE %s" % (pg_quote_identifier(self.name, 'publication'),
+                                                        pg_quote_identifier(table, 'table')))
+        if not check_mode:
+            return exec_sql(self, query, ddl=True)
```

Comment:
Why not pass `check_mode` into `exec_sql` and let it (if set to `True`) simply not execute the command? Then you don't have to repeat the pattern `if not check_mode: / exec_sql(...) / else: / self.executed_queries.append(...)` all over the module.

Reply:
Cool, thanks. I didn't change exec_sql itself because this is a shared function, so I added a new __exec_sql method to the class for these cases. It makes the code clear, agree, thanks!

Reply:
In case this pattern is used in other modules as well, moving it into `exec_sql` is probably a good idea. Also, adding a new argument to `exec_sql` with a default value shouldn't affect the other users.